### PR TITLE
Assignment of session 8

### PR DESCRIPTION
**Pull request by meta-shubham-jain**
The console is remaining to create. I will create and update again.

### CODE REVIEW

**Review by Ramkumar68 — CHANGES_REQUESTED** [review]

File: SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * It will add animal in Cage of Zone of Zoo
+     * @param name
+     * @param age
+     * @param weight
+     * @param typeOfAnimal
+     * @return
+     */
+    boolean addAnimal(String name, int age, double weight, String typeOfAnimal) {
```
[the remark this answers is not in the document]
Comment (Ramkumar68) [reply SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java]: Changes are not part of assignment

**Review by Ramkumar68 — APPROVED** [review]

File: SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		init();
+	}
+
+	public void init() {
+
+		zoneList.clear();
+		animalList.clear();
+		cageList.clear();
+
+		addZone(2, "Mammal", true, false);
```
Comment (Ramkumar68) [inline SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java]: @meta-shubham-jain 
Remove initialisation from class constructor. Create an object in JUnit class and add one by one.

File: SoftwareConstructionFundamental/SCF-Session8/JUnit.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.jupiter.api.Test;
+
+import zoo.Zoo;
+
+class JUnit {
+	
+	@Test
+    public void addZoneTest() {
+        // Adding a new Zone for Mammals which has Canteen and Park both
+        assertTrue(new Zoo().addZone(2, "Mammal", true, true));
```
Comment (Ramkumar68) [inline SoftwareConstructionFundamental/SCF-Session8/JUnit.java]: @meta-shubham-jain , Move to a variable.

**Review by Ramkumar68 — CHANGES_REQUESTED** [review]
Need to changes as in review

File: SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java (hunk 2 above)
Comment (meta-shubham-jain) [reply SoftwareConstructionFundamental/SCF-Session8/zoo/Zoo.java]: done

File: SoftwareConstructionFundamental/SCF-Session8/JUnit.java (hunk 3 above)
Comment (meta-shubham-jain) [reply SoftwareConstructionFundamental/SCF-Session8/JUnit.java]: Done

**Review by Ramkumar68 — APPROVED** [review]

**Pull request merged**